### PR TITLE
feat: replace uuid with prefixed string with schemaIdCounter

### PR DESCRIPTION
**Pull request by Uzlopak**
We dont need crypto randomUUID, because we are cool. We use an internal counter and a fsj specific prefic to generate the schema.id. 

@Ethan-Arrowood 
@inlet

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/uuid-v4.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const uuidV4 = require('../lib/uuid-v4')
+
+const uuidV4RE = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/u
+
+test('uuidV4', (t) => {
+  t.plan(2)
+  let alwaysLength36 = true
+  let validUUID4 = true
+  for (let i = 0; i < 1e6; ++i) {
+    const uuid = uuidV4()
```
[the remark this answers is not in the document]
Comment (Uzlopak) [reply test/uuid-v4.test.js:13]: yeah, we dont need cryptographical secure random number generation. We are just using the uuid for internal ids of the merged schemas etc..

[the remark this answers is not in the document]
Comment (jsumners) [reply test/uuid-v4.test.js:13]: I would be extremely wary of this. Maybe schema ids are not super secretive, but the _are_ meant to be globally unique and my uninformed guess is that `Math.random` is too predictable.

[the remark this answers is not in the document]
Comment (Uzlopak) [reply test/uuid-v4.test.js:13]: For our use case it should be good enough.

[the remark this answers is not in the document]
Comment (Uzlopak) [reply test/uuid-v4.test.js:13]: I dont know why github bot resolved this conversiation. I think @jsumners thought is important but still I think it is neglectable.

**Review by climba03003 — COMMENTED** [review]
Just curious, `crypto.randomUUID` is using pre-allocated cache.
Is it common to have that large of id generate which exceed the cache entropy?

**Comment by Uzlopak** [comment]
I guess it is for faster uuid generation. Like randomInt.

**Comment by mcollina** [comment]
If you need a unique identifier for just one process, you can use a global progressive integer.

File: test/uuid-v4.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Eomm) [reply test/uuid-v4.test.js:13]: Just to try out to solve this silly code:
`performance.now()` works on node and browses

**Comment by Uzlopak** [comment]
I personally think that collision wise it is easier to use this uuid implementation, because an incrementing number could result in collisions if multiple compiled json stringify are concentrated in one function. 

I think this is the cheapest solution, while also keeping it somewhat unique. I dont think that collisions are possible.

**Comment by mcollina** [comment]
> because an incrementing number could result in collisions if multiple compiled json stringify are concentrated in one function.

How? We don't support combining functions.

**Comment by Uzlopak** [comment]
I am worried that using just an incrementing number could result in some unexpected behavior in some dependent projects. Do we call then the internal schemas `__fjs_${schemaId}`

**Comment by ivan-tymoshenko** [comment]
Adding an incremental counter + internal prefix name should work.

**Comment by Uzlopak** [comment]
@mcollina 
@ivan-tymoshenko 

Implemented accordingly.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by ivan-tymoshenko — APPROVED** [review]

**Pull request merged**